plugins/inputs/beat/beat_test.go

```diff
@@ -0,0 +1,225 @@
+package beat
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/assert"
+)
+
+var beatTest = NewBeat()
+
+var (
+	beat6StatsAccumulator testutil.Accumulator
+)
+
+func Test_BeatStats(test *testing.T) {
+	//beat6StatsAccumulator.SetDebug(true)
+	fakeServer := httptest.NewUnstartedServer(
+		http.HandlerFunc(
+			func(writer http.ResponseWriter, request *http.Request) {
+				writer.Header().Set("Content-Type", "application/json")
+				if request.URL.String() == suffixInfo {
+					fmt.Fprintf(writer, "%s", string(beat6Info))
+				} else if request.URL.String() == suffixStats {
+					fmt.Fprintf(writer, "%s", string(beat6Stats))
+				} else {
+					test.Logf("Unkown URL: " + request.URL.String())
+				}
+			},
+		),
+	)
+	requestURL, err := url.Parse(beatTest.URL)
+	if err != nil {
+		test.Logf("Can't connect to: %s", beatTest.URL)
+	}
+	fakeServer.Listener, err = net.Listen("tcp", fmt.Sprintf("%s:%s", requestURL.Hostname(), requestURL.Port()))
+	if err != nil {
+		test.Logf("Can't listen for %s: %v", requestURL, err)
+	}
+
+	fakeServer.Start()
+	defer fakeServer.Close()
+
+	if beatTest.client == nil {
+		client, err := beatTest.createHttpClient()
+
+		if err != nil {
+			test.Logf("Can't createHttpClient")
+		}
+		beatTest.client = client
+	}
```

Comment:
Why is this necessary? The beat-plugin should connect on its own no?

Reply:
will fix